dlfcn_test.go

```diff
@@ -123,3 +112,43 @@ func TestNestedDlopenCall(t *testing.T) {
 
 	purego.Dlclose(lib)
 }
+
+func buildSharedLib(compilerEnv, libFile string, sources ...string) error {
+	out, err := exec.Command("go", "env", compilerEnv).Output()
+	if err != nil {
+		return fmt.Errorf("go env error: %w", err)
+	}
+
+	compiler := strings.TrimSpace(string(out))
+	if compiler == "" {
+		return errors.New("compiler not found")
+	}
+
+	args := []string{"-shared", "-Wall", "-Werror", "-o", libFile}
+
+	// macOS arm64 can run amd64 tests through Rossetta.
+	// Build the shared library based on the GOARCH and not
+	// the default behavior of the compiler.
+	var archFlag, arch string
+	if runtime.GOOS == "darwin" {
+		archFlag = "-arch"
+		switch runtime.GOARCH {
+		case "arm64":
+			arch = "arm64"
+		case "amd64":
+			arch = "x86_64"
+		default:
+			// if GOARCH is unknown then build the shared library
+			// with the compiler's normal target.
+			archFlag = ""
+			arch = ""
```

Comment:
Is this reachable? If not, what about returning an error?

Reply:
It probably wouldn't